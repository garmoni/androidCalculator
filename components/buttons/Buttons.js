import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import styles from './styles'

export default function Buttons({ setNum }) {
    const calcTable = []
    const btnCalc = ["AC","±","%","÷","mc","mr","m-","m+","7" ,"8" ,"9" ,"×","4" ,"5" ,"6" ,"-","1" ,"2" ,"3" ,"+","0" ,",","="]
    btnCalc.forEach((item, key) => {
      calcTable.push(
        <TouchableOpacity
          onPress={() => setNum(item)}
          title={item}
          key={key}
          style={[styles.Item, getStyle(item)]}
        ><Text style={styles.ItemText}>{item}</Text>        
        </TouchableOpacity>
      )
    });
    function getStyle(num) {
      switch (num){
        case "AC":
        case "±":
        case "%":
          return styles.Grey;
          break;
        case "m+":
        case "÷":
        case "×":
        case "-":
        case "+":
        case "=":
          return styles.Orange
          break;
        case "0" :
          return styles.bigItem
          break;
      }
    }
    return (
        <View style={styles.itemBlock}>{calcTable}</View>
      );
}