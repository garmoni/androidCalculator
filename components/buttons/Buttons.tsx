import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import styles from './styles'

type Props = {  setNum: Function, setLongPress: Function };

export default function Buttons({ setNum, setLongPress }:Props) {
    const btnCalc = ["AC","±","%","÷","mc","mr","m-","m+","7" ,"8" ,"9" ,"×","4" ,"5" ,"6" ,"-","1" ,"2" ,"3" ,"+","0" ,",","="]

    const getStyle = (num: string | undefined) => {
      switch (num){
        case "AC":
        case "±":
        case "%":
          return styles.Grey;
        case "m+":
        case "÷":
        case "×":
        case "-":
        case "+":
        case "=":
          return styles.Orange
        case "0" :
          return styles.bigItem
        default:
          break;
      }
    }
    return (
        <View style={styles.itemBlock}>
          {btnCalc.map((item: string, key: number) => (
              <TouchableOpacity
                onPress={() => setNum(item)}
                onLongPress={() => setLongPress(item)}
                key={key}
                style={[styles.Item, getStyle(item)]}
              >
                <Text style={styles.ItemText}>{item}</Text>        
              </TouchableOpacity>
          ))}
      </View>
      );
}
