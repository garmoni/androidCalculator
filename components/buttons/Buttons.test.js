import React from 'react';
import { render } from '@testing-library/react-native';
import Buttons from './Buttons'

const getStyle = jest.fn();

describe('Buttons', () => {

    it('should render Buttons Snapshot', () => {
        const container = render(<Buttons />)
        expect(container).toMatchSnapshot();
    });

    it('should work function getStyle', () => {
        render(<Buttons style={getStyle()}/>)
        expect(getStyle).toHaveBeenCalled()
    });
});
