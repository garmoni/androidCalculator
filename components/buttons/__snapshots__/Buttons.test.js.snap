// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons should render Buttons Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "space-around",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#a5a5a5",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      AC
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#a5a5a5",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      ±
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#a5a5a5",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      %
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ff980c",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      ÷
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#333",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      mc
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#333",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      mr
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#333",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      m-
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ff980c",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      m+
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#333",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      7
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#333",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      8
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#333",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      9
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ff980c",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      ×
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#333",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      4
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#333",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      5
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#333",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      6
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ff980c",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      -
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#333",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      1
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#333",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      2
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#333",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      3
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ff980c",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      +
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#333",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "paddingLeft": 25,
        "width": "46%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      0
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#333",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      ,
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ff980c",
        "borderRadius": 34,
        "display": "flex",
        "height": 68,
        "justifyContent": "center",
        "margin": "1%",
        "opacity": 1,
        "padding": 10,
        "width": 68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
      =
    </Text>
  </View>
</View>
`;
